src/main/java/com/example/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -32,6 +35,10 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
         User user = userMapper.toModel(requestDto);
         user.setPassword(passwordEncoder.encode(requestDto.password()));
         user.setRoles(Set.of(roleRepository.getByRole(Role.RoleName.USER)));
-        return userMapper.toDto(userRepository.save(user));
+        User savedUser = userRepository.save(user);
+        ShoppingCart cart = new ShoppingCart();
```

Comment:
It's better to create method createShoppingCart(User user) in ShopCartService and call it here